### ISSUE TITLE
kws_streaming bc_resnet does not stream

### ISSUE THREAD

**Issue by StuartIanNaylor** [google-research/google-research#1425]
@rybakov Hi Oleg, I thought I would give a bc_resnet a go on a custom dataset that has just run great on a crnn-state.
The end of test accuracy tests all fail with really strange results as if not streaming not sure how any pass.
Its the same for bc_resnet1 & bc_resnet2 on tensorflow 2.11.0
Only difference is the dataset and split
```
$CMD_TRAIN \
--data_url '' \
--data_dir $DATA_PATH/ \
--train_dir $MODELS_PATH/bc_resnet_1/ \
--split_data 0 \
--wanted_words 'heymarvin,noise,unk,h1m1,h2m1,h1m2,h2m2' \
--mel_upper_edge_hertz 7500 \
--mel_lower_edge_hertz 125 \
--how_many_training_steps 100,100,100,100,30000,30000,20000,10000,5000,5000 \
--learning_rate 0.001,0.002,0.003,0.004,0.005,0.002,0.0005,1e-5,1e-6,1e-7 \
--window_size_ms 30.0 \
--window_stride_ms 10.0 \
--mel_num_bins 40 \
--dct_num_features 0 \
--resample 0.0 \
--background_frequency 0.0 \
--alsologtostderr \
--train 1 \
--use_spec_augment 0 \
--time_masks_number 2 \
--time_mask_max_size 25 \
--frequency_masks_number 2 \
--frequency_mask_max_size 7 \
--pick_deterministically 1 \
bc_resnet \
--sub_groups 5 \
--last_filters 32 \
--first_filters 16 \
--paddings 'same' \
--dilations '(1,1),(2,1),(4,1),(8,1)' \
--strides '(1,1),(1,2),(1,2),(1,1)' \
--blocks_n '2, 2, 4, 4' \
--filters '8, 12, 16, 20' \
--dropouts '0.1, 0.1, 0.1, 0.1' \
--pools '1, 1, 1, 1' \
--max_pool 0
```
Here is a abridged version of the output.

```
I1218 10:14:47.303472 140365378379776 train.py:175] Step #100399: rate 0.000000, accuracy 86.00%, cross entropy 0.331316
I1218 10:14:47.729451 140365378379776 train.py:175] Step #100400: rate 0.000000, accuracy 74.00%, cross entropy 0.528566
I1218 10:16:26.224955 140365378379776 train.py:204] Step 100400: Validation accuracy = 88.01% (N=29400)
I1218 10:16:26.320242 140365378379776 train.py:222] So far the best validation accuracy is 88.04%
```
Lower than expected but hey.
```
I1218 10:16:26.325881 140365378379776 train.py:228] set_size=9800
I1218 10:17:02.699949 140365378379776 train.py:242] Final test accuracy = 88.06% (N=9800)
```
non stream tflite
```
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
I1218 10:19:41.592379 140365378379776 test.py:508] tflite test accuracy, non stream model = 87.56% 200 out of 9800
I1218 10:19:42.725630 140365378379776 test.py:508] tflite test accuracy, non stream model = 88.53% 400 out of 9800
I1218 10:19:43.856022 140365378379776 test.py:508] tflite test accuracy, non stream model = 87.19% 600 out of 9800
I1218 10:19:44.987253 140365378379776 test.py:508] tflite test accuracy, non stream model = 86.39% 800 out of 9800
```

```
 This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.
I1218 10:23:40.933223 140365378379776 test.py:309] tf test accuracy, stream model state external = 11.94% 200 out of 1000
I1218 10:26:33.526979 140365378379776 test.py:309] tf test accuracy, stream model state external = 12.97% 400 out of 1000
I1218 10:29:28.372378 140365378379776 test.py:309] tf test accuracy, stream model state external = 13.48% 600 out of 1000
I1218 10:32:23.397092 140365378379776 test.py:309] tf test accuracy, stream model state external = 13.61% 800 out of 1000
I1218 10:35:15.902516 140365378379776 test.py:314] TF Final test accuracy of stream model state external = 14.70% (N=1000)
```

```
I1218 10:38:21.621161 140365378379776 test.py:309] tf test accuracy, stream model state external = 16.42% 200 out of 1000
I1218 10:41:17.279755 140365378379776 test.py:309] tf test accuracy, stream model state external = 16.21% 400 out of 1000
I1218 10:44:13.251102 140365378379776 test.py:309] tf test accuracy, stream model state external = 15.31% 600 out of 1000
I1218 10:47:06.414232 140365378379776 test.py:309] tf test accuracy, stream model state external = 15.36% 800 out of 1000
I1218 10:50:00.302742 140365378379776 test.py:314] TF Final test accuracy of stream model state external = 15.10% (N=1000)
```

```
I1218 10:53:12.230915 140365378379776 test.py:173] tf test accuracy, stream model state internal = 13.43% 200 out of 1000
I1218 10:56:05.987689 140365378379776 test.py:173] tf test accuracy, stream model state internal = 12.47% 400 out of 1000
I1218 10:58:57.284193 140365378379776 test.py:173] tf test accuracy, stream model state internal = 12.98% 600 out of 1000
I1218 11:01:50.017224 140365378379776 test.py:173] tf test accuracy, stream model state internal = 13.48% 800 out of 1000
I1218 11:04:42.984909 140365378379776 test.py:178] TF Final test accuracy of stream model state internal = 13.60% (N=1000)
I1218 11:04:43.203384 140365378379776 model_train_eval.py:266] run TFlite streaming model accuracy evaluation
```

```
I1218 11:06:15.920165 140365378379776 test.py:433] tflite test accuracy, stream model state external = 14.420620 9000 out of 9800
I1218 11:06:17.768203 140365378379776 test.py:433] tflite test accuracy, stream model state external = 14.476687 9200 out of 9800
I1218 11:06:19.594013 140365378379776 test.py:433] tflite test accuracy, stream model state external = 14.434635 9400 out of 9800
I1218 11:06:21.456112 140365378379776 test.py:433] tflite test accuracy, stream model state external = 14.352672 9600 out of 9800
I1218 11:06:23.295054 140365378379776 test.py:438] tflite Final test accuracy, stream model state external = 14.29% (N=9800)

```

```
I1218 11:07:45.612886 140365378379776 test.py:433] tflite test accuracy, stream model state external = 14.271106 8800 out of 9800
I1218 11:07:47.389667 140365378379776 test.py:433] tflite test accuracy, stream model state external = 14.242862 9000 out of 9800
I1218 11:07:49.193121 140365378379776 test.py:433] tflite test accuracy, stream model state external = 14.313662 9200 out of 9800
I1218 11:07:50.968217 140365378379776 test.py:433] tflite test accuracy, stream model state external = 14.317626 9400 out of 9800
I1218 11:07:52.759829 140365378379776 test.py:433] tflite test accuracy, stream model state external = 14.227685 9600 out of 9800
I1218 11:07:54.592904 140365378379776 test.py:438] tflite Final test accuracy, stream model state external = 14.29% (N=9800)
I1218 11:07:54.644539 140365378379776 model_train_eval.py:203] feature type mfcc_tf needs quantization aware training for quantization - it is not implemented
```

Is it tf2.11.0 and we are getting to the end of the v1 api? As sort of confused if not streaming how is it getting any results on the streaming tests but its also out of whack with the training results whilst non-stream seems OK just not great with the dataset I had?

**Comment by StuartIanNaylor** [comment]
Apols found it

https://github.com/google-research/google-research/issues/1158#issuecomment-1167895578

**Comment by rybakov** [comment]
To run this model in streaming mode you need to set:
'''
paddings 'causal' \
'''

You can stream non causal models with 
'''
--paddings 'same'
'''
but you will need to modify the model and introduce delay as in [example](https://github.com/google-research/google-research/blob/master/kws_streaming/layers/delay_test.py#L106)
